boards/DCM/Test/Src/Test_StateMachine.cpp

```diff
@@ -399,4 +420,169 @@ TEST_F(
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_RIGHT_WHEEL_SPEED(
+        can_rx_interface, threshold_wheel_speed);
+    LetTimePass(state_machine, 10);
+    AssertFloatsEqual(
+        expected_torque_request_value,
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+
+    // Check regen doesn't turn on when both wheels are going faster than 5kph
+    // but one of the AIRs is open
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_LEFT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_RIGHT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_OPEN_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+    AssertFloatsEqual(
+        expected_torque_request_value,
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_OPEN_CHOICE);
+    LetTimePass(state_machine, 10);
+    AssertFloatsEqual(
+        expected_torque_request_value,
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+
+    // Check regen turns on when both wheels are going faster than 5kph and AIRS
+    // are closed
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+    AssertFloatsEqual(
+        expected_regen_request_value,
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+}
+
+// DCM-19
+TEST_F(DcmStateMachineTest, torque_requests_faster_than_100Hz)
+{
+    float torque_request_with_regen_allowed;
+    float torque_request_without_regen_allowed;
+
+    SetInitialState(App_GetDriveState());
+
+    // Turn the DIM start switch on to prevent state transitions in
+    // the drive state.
+    EXPECT_TRUE(App_BuzzerSignals_IsOn(world));
+    App_CanRx_DIM_SWITCHES_SetSignal_START_SWITCH(
+        can_rx_interface, CANMSGS_DIM_SWITCHES_START_SWITCH_ON_CHOICE);
+    ASSERT_EQ(
+        App_GetDriveState(),
+        App_SharedStateMachine_GetCurrentState(state_machine));
+
+    // Check that no torque requests are sent when the accelerator pedal is not
+    // pressed
+    // ("The DCM may only request torque less than or equal to what the driver
+    // requested")
+    LetTimePass(state_machine, 10);
+    AssertFloatsEqual(
```

Comment:
why is AssertsFloatsEqual necessary

Reply:
The tests don't pass because 15.9999f != 16f

Reply:
I prefer we just stick to the regular ASSERT_EQ or ASSERT_FLOAT_EQ

In this particular line of code AssertFloatsEqual shouldn't be necessary. If ASSERT_FLOAT_EQ still doesn't work then could you select a mapped pedal percentage? (preferably a simpler one like 50%)

Reply:
Lol I just didn't know `ASSERT_FLOAT_EQ` was a thing